### PR TITLE
Transpile to ES5 (Fixes: #43, #41, #24, #23)

### LINKED ISSUE #43: [BUG] Zipcelx make my Uglify crash
Hey guys,

First and foremost, thanks you for the amazing work you did providing us this small and super usefull library. I was totally delighted when I found it (thanks to medium).

So I used it in our Angular application, and everythings ok, except when it came to minimize the code we're shipping.

I get the following error during production build : 
```
ERROR in ./dist/0.7da5d6ef.js from UglifyJs
Unexpected token: operator (>) [./dist/0.7da5d6ef.js:60268,32]
```

After investigation, it seems that Uglify doesn't support arrow function yet 😭 

I saw in #31 and #24 and #23 that there are others people wihch seems to have the same problem.

So far I've tried :
```js
// Prod compilation crash
import zipcelx from 'zipcelx';
// Prod compilation crash
import 'zipcelx';
// Prod compilation crash
import { zipcelx } from 'zipcelx';
// Prod compilation workgin but crahs in the console when trying to use it
import zipcelx from 'zipcelx/lib/standalone';
```

But no one is working. **So what is the correct way to use zipcelx without crashing Uglify ?**

Thanks :)

### PR DESCRIPTION
**Pull request by mwld**
Hi there,

thanks for the great plugin! It does a neat job in generating .xlsx files from simple data and it’s so much smaller than [js-xlsx](https://github.com/SheetJS/js-xlsx/)!

Nevertheless I need libraries to be ES5-compatible (for IE11 support etc.). That’s why I added [rollup-plugin-babel](https://github.com/rollup/rollup-plugin-babel) to the build process. This way module and standalone versions will be compatible with browsers that are not supporting ES2015 (yet).

Fixes #43, #41, #24 and #23.

PS: One might argue that it’s not the libraries job to transpile to ES5 but the one of the build process including the library. In my opinion packages made for the web should support current web technology out there which includes some older browsers. The authors of _rollup-plugin-babel_ seem to think similar: 
> We encourage library authors not to distribute code that uses untranspiled ES6 features (other than modules) for this reason. Consumers of your library should not have to transpile your ES6 code, any more than they should have to transpile your CoffeeScript, ClojureScript or TypeScript.
(https://github.com/rollup/rollup-plugin-babel#external-dependencies)

### CODE REVIEW

**Comment by egeriis** [comment]
Thank you for bringing this up, I'm glad to talk about this.

There are unfortunately no set standard for whether or not libs should be distributed as ES5 og ES6. However we've decided to follow the guidelines outlined by Dan Abramov here: 
https://twitter.com/dan_abramov/status/1009179550134296577

tl;dr: don't ship experimental features.

The main purpose being to keep our output file as small as possible. Users who do not need to support legacy browsers, will otherwise have a lot of boilerplate that are not needed as part of their build.

That being said, I am not confident that our current build process strips experimental code. So that's one action items. This is also not set in stone, and I would love to debate what the appropriate output for a library is.

---

>  it’s so much smaller than js-xlsx!

This lib's purpose exactly!

**Comment by mwld** [comment]
I think it comes down to the question which browsers an end user library should support.

I consider this library being made for the normal web user as it generates XLSX files. And those will probably be opened in Microsoft Excel. So most of the consumers might use it on a Windows machine which uses Internet Explorer or Edge as default browser (in a corporate environment). But this is just speculation.

The guys from [compat-table](https://kangax.github.io/compat-table/es6/) had a similar discussion and came to the conclusion to support browsers which have not reached their end of lifecycle yet: https://github.com/kangax/compat-table/pull/1114#issuecomment-310062383

`The main purpose being to keep our output file as small as possible. Users who do not need to support legacy browsers, will otherwise have a lot of boilerplate that are not needed as part of their build.`

You are right that this should be kept in mind. In this specific case the output sizes increased by 0.3 KB (module.js) and 1 KB (standalone.js). Which is acceptable in my opinion compared to the benefit it provides. (E.g. for the authors of #43, #41, #24 and #23.)

**Comment by egeriis** [comment]
> I consider this library being made for the normal web user as it generates XLSX files. And those will probably be opened in Microsoft Excel.

Yeah, I agree it's speculation. But I also think this is irrelevant.

We have the standalone build specifically for consumers who are not using a build system. The other build is intended to be used in environments where the consumers have a build system, and thus we want to give them full control over the output of their build. I.e. we ship non-experimental ES6+ features in our build, if needed, transpile these according to own best practices at your own discretion.

**Comment by egeriis** [comment]
A bit more thought on the actions to take here. We should probably add `module` build back, which should point to the current `main` build. `main` should be CJS https://rollupjs.org/guide/en#compatibility

We had this before, but I removed it. Don't remember why and cannot find any conversation around it.

Add babel transforms so we only ship ES proposals that has been implemented. I think using the [stage-2](https://babeljs.io/docs/en/babel-preset-stage-2) preset is a sane option.

**Comment by emorling** [comment]
@egeriis @doliG whats the status on this? When will the zipcelx-es5-cjs be merged?

**Comment by doliG** [comment]
Don't think so. Currently I'm using a [fork](https://yarnpkg.com/fr/package/zipcelx-es5-cjs) I've created to solve the problem.
I think I'm gonna create a PR to merge all this stuff but dunno when :)

**Comment by egeriis** [comment]
Stalled

**Comment by emorling** [comment]
@doliG can we get an update?

**Comment by egeriis** [comment]
@emorling This should be fixed in stable as of  https://github.com/egeriis/zipcelx/pull/56
